File: 0328-odd-even-linked-list/0328-odd-even-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var oddEvenList = function(head) {
    let newList = new ListNode();
    let evenList = new ListNode();
    let answer = newList;
    let evenHead = evenList;
    let current = head;

    let count = 1;
    while(current){
        if(count%2 === 1){
            newList.next = current;
            newList = newList.next;
        } else {
            evenList.next = current;
            evenList = evenList.next;
        }
        current = current.next;
        count++
    }

    newList.next = evenHead.next;
    evenList.next = null;


    return answer.next;
};

File: 0328-odd-even-linked-list/0328-odd-even-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def oddEvenList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        newList = ListNode()
        evenList = ListNode()
        answer=newList
        headOfEven = evenList

        current = head

        count = 1

        while current:
            if count % 2 == 1:
                newList.next = current
                newList = newList.next
            else:
                evenList.next = current
                evenList = evenList.next
            current = current.next
            count += 1
        
        newList.next = headOfEven.next
        evenList.next = None

        return answer.next